tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -235,7 +238,7 @@ class GUnmarshalSum f where
                    => m (f a)
 
 instance (Unmarshal k, SymbolMatching k) => GUnmarshalSum (M1 C c (M1 S s (K1 i k))) where
-  gunmarshalSumNode = M1 . M1 . K1 <$> unmarshalNode
+  gunmarshalSumNode = M1 . M1 . K1 <$> (peekNode >>= unmarshalNodes . maybeToList)
```

Comment:
Similar to what you did with `gunmarshalNode`, I think `gunmarshalSumNode` could take the current node as well and save a redundant call to `peekNode`.